### PR TITLE
Content page component design fixes

### LINKED ISSUE #1090: Create `VContentPage` component
## Problem

When building text-heavy content pages, it's difficult to make sure the styles are correct on every page. To fix this, we could build a set of typography primitives: `<Heading level="1">` type components, and so on. Until we get to that stage, which would require a solid amount of work, it would be easier to create a `VContentPage` component that wraps raw html tags and applies styles to them using https://tailwindcss.com/docs/typography-plugin

This will allow us to make copy edits without also having to write many complex styles, and allow our content pages to be consistent by default. Then, if and when we make standalone small typography components, we can 'turn off' features of the tailwind plugin and switch to those incrementally.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1090 by @zackkrida

## Description

This PR attempts to finish up the work started in #1093 to get it merged :)
It merges main and applies the following #1093 code review suggestions:

**On desktop**
- Margin bottom of h1 set to 40px.
- Margin top of h3 set to 40px.
- Links in main content only have an underline style on hover.
- In the table, labels in header cells are description bold style.
- In the table, texts in cells are description regular style.
- External icon in main CTA in sources uses correct icon.

**On mobile**
- h1, h3 styles updated

#### The following suggestions were not implemented:
- Mobile paragraph styles not updated.
- The table layout on mobile is unchanged.

#### Other changes:
- `scss` removed from the pages styles
- `VContentPage` name added so that it can be imported in the components.

## Testing Instructions

Check that the content pages look similar to the Sources page mockup, and are not broken.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VContentPage.vue [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      <slot default />
+    </div>
+  </div>
+</template>
+<script>
+const VContentPage = {
+  name: 'VContentPage',
+}
+export default VContentPage
+</script>
```
Comment (sarayourfriend) [inline src/components/VContentPage.vue:15]: Take it or leave it :slightly_smiling_face: 

```suggestion

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api'
export default defineComponent({
  name: 'VContentPage',
})
</script>
```

File: tailwind.config.js [hunk 2]
```diff
@@ -202,7 +202,26 @@ module.exports = {
...
+            '--tw-prose-links': theme('colors.pink'),
+            a: {
+              textDecoration: 'none',
+              '&:hover': {
+                'text-decoration': 'underline',
+              },
+            },
+          },
+        },
+      }),
```
Comment (sarayourfriend) [inline tailwind.config.js:219]: Isn't this redundant with what we have in `tailwind.css` in the `@base` layer extension?

**Review by sarayourfriend — COMMENTED** [review]
Tested locally and looks good to me. One question before approving but otherwise looks good.

File: src/components/VContentPage.vue (hunk 1 above)
Comment (zackkrida) [reply src/components/VContentPage.vue:15]: Nice catch @sarayourfriend, I found myself staring at this in my review thinking "hm, something isn't right here" 😆

File: tailwind.config.js (hunk 2 above)
Comment (obulat) [reply tailwind.config.js:219]: With the new tailwind plugin, typography, the texts and links on the content pages have different classes applied to them, so it becomes necessary.

File: package.json [hunk 3]
```diff
@@ -58,6 +58,7 @@
     "@pinia/nuxt": "^0.1.8",
     "@popperjs/core": "^2.11.2",
     "@tailwindcss/line-clamp": "^0.3.1",
+    "@tailwindcss/typography": "^0.5.2",
```
Comment (sarayourfriend) [inline package.json:61]: I guess I don't fully understand why we're using this plugin and why if we _do_ use it it doesn't render the code in `tailwind.css` unnecessary (should we remove it if we use this plugin?).

File: src/components/VContentPage.vue [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<template>
+  <div class="pt-5 md:pt-10" dir="ltr">
+    <div
+      class="px-6 lg:px-0 mb-10 lg:mb-30 md:max-w-4xl xl:max-w-5xl prose prose-sm md:prose-base mx-auto max-w-none prose-headings:font-bold lg:prose-headings:text-3xl prose-h1:text-4xl prose-h1:text-bold lg:prose-h1:text-6xl md:prose-h1:mb-10 prose-h3:text-2xl prose-h3:font-semibold md:prose-h3:mt-10"
```
Comment (sarayourfriend) [inline src/components/VContentPage.vue:4]: This massive list of `prose-*` classes also scares me. It has all the features of the worst parts of Tailwind IMO (namely, a super long and inscrutable list of class names that are difficult to edit, read, and get a full picture of).

Adding our own `prose` class to `tailwind.css` with these styles using the `@apply` rule would probably be easier to read and maintain :shrug: Just my own feeling about this plugin though.

File: package.json (hunk 3 above)
Comment (sarayourfriend) [reply package.json:61]: Makes sense, thanks!

**Comment by zackkrida** [comment]
@sarayourfriend the primary reason I'd like to use the plugin is so we have more raw html tags in use and can easily replace things, like `<h1>` ~> `<Heading level="1">` or whatever if and when we make our design system more robust, instead of having to do a bunch of manual work or, god forbid, regex matching on html tags with classes and other attributes. Your thoughts are well-received!

**Pull request merged**